### ISSUE TITLE
Vue <Transition appear> causing Hydration mismatch on Astro 4.1.1

### ISSUE THREAD

**Issue by sph-e** [withastro/astro#9636]
### Astro Info

```block
Astro                    v4.1.1
Node                     v18.18.0
System                   Linux (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             @astrojs/vue
```

### Describe the Bug

Using `<Transition appear>` causes hydration issues, there are no issues without the "appear" keyword,

Bug doesnt appear to exist in 

"@astrojs/vue": "^4.0.3",
"astro": "^4.0.4",

And lower, started happening when upgraded to 4.1.1

To reproduce: Run the minimal reproduction, and then change something in a file and hit save. That should reload the page and trigger the hydration mismatch

![image](https://github.com/withastro/astro/assets/114557048/11b79a90-e417-49c8-b26b-7645f9ee5501)

### What's the expected result?

Server shouldn't be attempting to put [object Set] as the class

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-pffhpt?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by florian-lefebvre** [comment]
I can confirm it still happens on latest:
```json
"@astrojs/vue": "^4.0.8",
"astro": "^4.2.4",
"vue": "^3.4.15"
```

**Comment by mingjunlu** [comment]
I'm not familiar with Vue, but here's what I discovered:

- The `[object Set]` problem is resolved after upgrading to `vue@^3.4.8`. However, the hydration mismatch warning in the minimal reproducible example still exists even after upgrading Vue to the latest version
- I tried to put the Vue component into an isolation environment without Astro, using the [Vue SFC playground](https://play.vuejs.org/#__SSR__eNqNUstOwzAQ/BXLFy40La9LCUiAOMABEHCMhCxnKS6Obdmb0Krqv7O226YgWnHz7sw+ZtYLfuVc0bXAx7wM0iuHLAC2jmlhJhcVx1Dxy8qoxlmPbMFevTBBobKGLdm7tw07oOqD88qUw1xPbAoQGqcFAkWMlbXq0oOefYOcYAwMgh8IiaqDgdQihDi3p/GfRLQ9yfagBkHlcaMNHIPfhH+MycQdY4RzIHyOVpKyPrZmBxSoZC4m78rhlvhhrz4bswLL4ZZfFAaca2BBWgc1ZYp+TbaIdX08qFtP86wZs5PRqAl0iGVsUaTbJLJ1Qiqcj9moONvAaHeBdMc4ncbyQzq/tOZdTYppsIb+SKqpuLSNUxr8o4ujSeY4d4uY0Np+3acc+hYO13n5AfLzj/w0zGKu4k8eAvgOKr7BUPgJYIZvXx5gRu8N2Ni61cTeAz5DsLrN542069bUtPYWL217l363MpPXcDtDIGtXouKikblM/IrTV7/ZI71f96Q4TXXkKLn41oGPPclAAorjI778BuA/JhY=) with SSR on. An error message about hydration mismatch still showed up: `Hydration completed but contains mismatches.`

Would it be possible that this issue is more related to Vue instead of Astro?

---

Possibly related issues:

- https://github.com/vuejs/core/issues/10067
- https://github.com/vuejs/core/issues/6951
